content/docs/cucumber/configuration.md

```diff
@@ -15,170 +15,376 @@ polyglot:
 # Type Registry
 
 {{% block "java,kotlin" %}}
-The type registry is used to configure parameter types and data table types. It can be configured by placing an implementation
-of `cucumber.api.TypeRegistryConfigurer` on the glue path.
+Parameter types let you convert parameters from cucumber-expressions to objects.
+Data table and doc string string types let you convert data tables and doc
+strings to objects. Like step definitions, type definitions are part of the glue.
+When placed on the glue path Cucumber will detect them automatically. 
```

Comment:
Do we explain "glue" and "glue path" anywhere? Am thinking a glossary might be helpful (probably outside of the scope of this PR)